--- ./stonith/Makefile.org	Sat Sep 29 21:55:25 2001
+++ ./stonith/Makefile	Mon Sep  2 13:54:30 2002
@@ -156,10 +156,10 @@
 	$(INSTALL) -cm 755 $(BINCMDS) $(BINDIR)
 	ls -l $(BINDIR)/$(BINCMDS)
 
-install_lib: $(LIBFILES) $(MODULES) baytech.so rps10.so nw_rpc100s.so $(VACM_STONITH)
+install_lib: $(LIBFILES) $(MODULES) $(VACM_STONITH)
 	$(INSTALL) -cm 755 $(LIBFILES) $(LIBDIR)
 	mkdir -p $(MODDIR)
-	$(INSTALL) -cm 755 $(MODULES:.o=.so) baytech.so rps10.so nw_rpc100s.so $(VACM_STONITH) $(MODDIR)
+	$(INSTALL) -cm 755 $(MODULES:.o=.so) $(VACM_STONITH) $(MODDIR)
 
 install_inc:
 	if [ ! -d $(INCDIR) ]; then mkdir $(INCDIR); fi
