--- heartbeat-1.99.5/lib/clplumbing/GSource.c.orig	2005-05-24 07:44:45.000000000 +0200
+++ heartbeat-1.99.5/lib/clplumbing/GSource.c	2005-06-14 15:08:46.000000000 +0200
@@ -988,6 +988,12 @@
 	if (pid < 0 && errno != ECHILD) {
 		cl_perror("%s: wait3() failed"
 		,	__FUNCTION__);
+#if defined(DEBUG)
+		if (childcount < 1 && ANYDEBUG) {
+			cl_log(LOG_DEBUG, "%s called without children to wait on"
+			,	__FUNCTION__);
+		}
+#endif
 	}
 	if (alarm_count) {
 		cl_log(LOG_ERR
