diff -urNp -x '*.orig' Heartbeat-3-0-STABLE-3.0.6.org/configure.ac Heartbeat-3-0-STABLE-3.0.6/configure.ac
--- Heartbeat-3-0-STABLE-3.0.6.org/configure.ac	2015-02-04 15:57:04.000000000 +0100
+++ Heartbeat-3-0-STABLE-3.0.6/configure.ac	2024-04-25 23:59:40.319018426 +0200
@@ -777,18 +777,7 @@ fi
 dnl Determine if we use -w1 or -t1 for ping (in PYTHON Scripts)
 
 
-AC_MSG_CHECKING(ping option for timeout - w or t)
-for PING_TIMEOUT_OPT in "-w1" "-t1"
-do
-  $PING -nq -c1 $PING_TIMEOUT_OPT localhost > /dev/null 2>&1
-  if
-    test "$?" = 0
-  then
-    AC_DEFINE_UNQUOTED(PING_TIMEOUT_OPT, "$PING_TIMEOUT_OPT", option for ping timeout)
-    AC_MSG_RESULT($PING_TIMEOUT_OPT)
-    break
-  fi
-done
+PING_TIMEOUT_OPT="-w1"
 AC_SUBST(PING_TIMEOUT_OPT)
 
 
@@ -806,17 +795,7 @@ AC_DEFINE(PROCROUTE, "/proc/net/route",
 
 
 AC_MSG_CHECKING(ifconfig option to list interfaces)
-for IFCONFIG_A_OPT in  "-A" "-a" ""
-do
-  $IFCONFIG $IFCONFIG_A_OPT > /dev/null 2>&1
-  if
-    test "$?" = 0
-  then
-    AC_DEFINE_UNQUOTED(IFCONFIG_A_OPT, "$IFCONFIG_A_OPT", option for ifconfig command)
-    AC_MSG_RESULT($IFCONFIG_A_OPT)
-    break
-  fi
-done
+IFCONFIG_A_OPT=""
 
 AC_SUBST(IFCONFIG_A_OPT)
 AC_SUBST(WHOAMI)
@@ -1585,8 +1564,12 @@ dnl ************ printw ****************
 	      [
 #if defined(HAVE_CURSES_H)
 #  include <curses.h>
+#elif defined(HAVE_CURSES_CURSES_H)
+#  include <curses/curses.h>
 #elif defined(HAVE_NCURSES_H)
 #  include <ncurses.h>
+#elif defined(HAVE_NCURSES_NCURSES_H)
+#  include <ncurses/ncurses.h>
 #endif
 	      ],
 	      [printw((const char *)"Test");]
