--- heartbeat/Makefile.orig	Thu Apr  4 19:51:25 2002
+++ heartbeat/Makefile	Thu Apr  4 19:51:49 2002
@@ -8,7 +8,7 @@
 #
 PKG=heartbeat
 
-INITD=/etc/init.d
+INITD=$(shell [ -d /etc/init.d ] && echo /etc/init.d || echo /etc/rc.d/init.d )
 LOGROTATED=$(BUILD_ROOT)/etc/logrotate.d
 
 #       Debian wants things to start with DESTDIR,
--- heartbeat/heartbeat.sh.orig	Thu Apr  4 19:53:18 2002
+++ heartbeat/heartbeat.sh	Thu Apr  4 19:53:23 2002
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 #	$Id$
 #
