--- heartbeat-1.99.5/lib/plugins/stonith/ipmilan_command.c.orig	2005-01-31 10:45:31.000000000 +0100
+++ heartbeat-1.99.5/lib/plugins/stonith/ipmilan_command.c	2005-06-03 11:20:41.000000000 +0200
@@ -297,7 +297,7 @@
 		return S_ACCESS;
 	}
 
-	con->set_con_change_handler(con, con_changed_handler, &request);
+	con->add_con_change_handler(con, con_changed_handler, &request);
 
 	gstatus = IPMI_RUNNING;
 
