--- heartbeat-1.99.5/configure.in.orig	2005-05-24 07:20:02.000000000 +0200
+++ heartbeat-1.99.5/configure.in	2005-06-03 11:08:58.000000000 +0200
@@ -746,9 +746,9 @@
 AC_SUBST(SNMPCONFIG)
 AC_SUBST(SNMPLIB)
 
-if test x"$VERSION" \> x"1.9"; then
+#if test x"$VERSION" \> x"1.9"; then
 	AC_DEFINE(HAVE_NEW_HB_API, 1, [have new heartbeat api])
-fi
+#fi
 
 dnl
 dnl     On many systems libcrypto is needed when linking against libsnmp.
@@ -1747,7 +1747,7 @@
 			FatalMissingThing "swig" "SWIG version must be > 1.3.19" \
 			"You can get it from http://swig.org/"
 		else
-			if $PERL -e "use 5.008001;" >& /dev/null; then
+			if $PERL -e "use 5.008001;" >/dev/null 2>&1; then
 				SWIG_ENABLED=1
 				AC_DEFINE_UNQUOTED(WITH_SWIG, 1, 
 						SWIG bindings enabled)
--- heartbeat-1.99.5/config/snmp_subagent.m4.orig	2004-09-28 08:39:38.000000000 +0200
+++ heartbeat-1.99.5/config/snmp_subagent.m4	2005-06-03 11:46:10.000000000 +0200
@@ -209,7 +209,7 @@
 	    dnl test net-snmp version
 	    NET_SNMP_VERSION=`net-snmp-config --version`
 	    AC_MSG_WARN([NET-SNMP: verion = $NET_SNMP_VERSION])
-	    if test x"$NET_SNMP_VERSION" \< x"5.1" ; then
+	    if test "$NET_SNMP_VERSION" -lt "5.1" ; then
 		AC_MSG_ERROR([NET-SNMP: you need NET-SNMP 5.1 or above. ])
 	    fi
 
