--- Heartbeat-3-0-STABLE-3.0.6/configure.ac.orig	2015-12-19 16:20:41.745674168 +0100
+++ Heartbeat-3-0-STABLE-3.0.6/configure.ac	2015-12-19 16:47:03.728941113 +0100
@@ -624,6 +624,7 @@
 fi
 AC_SUBST(DLOPEN_FORCE_FLAGS)
 
+LIBUUID=
 dnl ************ uuid **********************
 AC_ARG_WITH(uuid,
     [  --with-uuid=UUID      mechanism for uuid {e2fsprogs|ossp|any|no}
@@ -636,25 +637,26 @@
   )
 case $uuidimpl in
 e2fsprogs)
-  AC_CHECK_LIB(uuid, uuid_parse,,
+  AC_CHECK_LIB(uuid, uuid_parse,[LIBUUID=-luuid],
     AC_MSG_ERROR([e2fsprogs uuid library was explicitly requested but not found])
   )
   ;;
 ossp)
-  AC_CHECK_LIB(uuid, uuid_create,,
+  AC_CHECK_LIB(uuid, uuid_create,[LIBUUID=-luuid],
     AC_MSG_ERROR([ossp uuid library was explicitly requested but not found])
   )
   ;;
 no)	# Do not define anything; so use inbuilt (replace) implementation.
   ;;
 any)	# Default: try to discover an implementation that we can use.
-  AC_CHECK_LIB(uuid, uuid_parse)	dnl e2fsprogs
+  AC_CHECK_LIB(uuid, uuid_parse,[LIBUUID=-luuid])	dnl e2fsprogs
 dnl  AC_CHECK_LIB(uuid, uuid_create)	dnl ossp
   ;;
 *)
   AC_MSG_ERROR([An invalid uuid library was explicitly requested])
   ;;
 esac
+AC_SUBST([LIBUUID])
 
 case "$host_os" in
 darwin*)	
@@ -1892,7 +1894,7 @@
 dnl	getopt: If this is valid, define HAVE_DECL_GETOPT to make the getopt.h header compile cleanly.
 dnl
 
-AC_REPLACE_FUNCS(alphasort inet_pton NoSuchFunctionName scandir setenv strerror unsetenv strnlen strndup daemon uuid_parse strlcpy strlcat)
+AC_REPLACE_FUNCS(alphasort inet_pton NoSuchFunctionName scandir setenv strerror unsetenv strnlen strndup daemon strlcpy strlcat)
 dnl AC_CHECK_FUNCS(alphasort inet_pton NoSuchFunctionName scandir setenv strerror unsetenv strnlen strndup daemon uuid_parse)
 
 AC_CHECK_FUNCS(getopt, 
@@ -1991,7 +1993,7 @@
 dnl check if header file and lib are there for zlib
 zlib_installed="yes"
 AC_CHECK_HEADERS(zlib.h, , [zlib_installed="no"],)
-AC_CHECK_LIB(z, compress , , [zlib_installed="no"])
+AC_CHECK_LIB(z, compress , [LIBZ=-lz] , [zlib_installed="no"])
 AM_CONDITIONAL(BUILD_ZLIB_COMPRESS_MODULE, test "x${zlib_installed}" = "xyes")
 if test "x${zlib_installed}" = "xno"; then
 	FatalMissingThing "zlib" \
@@ -2002,7 +2004,7 @@
 dnl check if header file and lib are there for zlib
 bz2_installed="yes"
 AC_CHECK_HEADERS(bzlib.h, , [bz2_installed="no"],)
-AC_CHECK_LIB(bz2, BZ2_bzBuffToBuffCompress , , [bz2_installed="no"])
+AC_CHECK_LIB(bz2, BZ2_bzBuffToBuffCompress , [LIBBZ2=-lbz2], [bz2_installed="no"])
 AM_CONDITIONAL(BUILD_BZ2_COMPRESS_MODULE, test "x${bz2_installed}" = "xyes")
 
 dnl check if there are getpid() inconsistency
@@ -2118,7 +2120,7 @@
 	GNUTLSLIBS="`$LIBGNUTLS_CONFIG --libs`";
 	AC_MSG_RESULT($GNUTLSLIBS)
 else
-	AC_CHECK_LIB(gnutls, gnutls_init)
+	AC_CHECK_LIB(gnutls, gnutls_init, [GNUTLSLIBS=-lgnutls])
 fi
 AC_SUBST(GNUTLSHEAD)
 AC_SUBST(GNUTLSLIBS)
