--- heartbeat-3.0.2-rc2/contrib/drbd-outdate-peer/Makefile.am.orig	2011-04-27 15:06:45.276677110 +0200
+++ heartbeat-3.0.2-rc2/contrib/drbd-outdate-peer/Makefile.am	2011-04-27 15:06:52.406677078 +0200
@@ -39,7 +39,7 @@
 			$(top_builddir)/lib/hbclient/libhbclient.la $(GLIBLIB)
 
 drbd_peer_outdater_SOURCES	= drbd-peer-outdater.c
-drbd_peer_outdater_LDADD	= -lplumb
+drbd_peer_outdater_LDADD	= -lplumb $(GLIBLIB)
 
 
 
