--- stonith/expect.c	
+++ stonith/expect.c
@@ -38,6 +38,7 @@
 #include <errno.h>
 #include <sys/time.h>
 #include <sys/times.h>
+#include <time.h>
 #ifdef _POSIX_PRIORITY_SCHEDULING
 #	include <sched.h>
 #endif
--- heartbeat/hb_api.c	
+++ heartbeat/hb_api.c	
@@ -76,6 +76,7 @@
 #include <sys/stat.h>
 #include <sys/time.h>
 #include <unistd.h>
+#include <time.h>
 
 static int api_ping_iflist(const struct ha_msg* msg, struct node_info * node
 ,	struct ha_msg* resp
