--- heartbeat-1.99.5/lib/clplumbing/GSource.c.orig	2005-05-24 07:44:45.000000000 +0200
+++ heartbeat-1.99.5/lib/clplumbing/GSource.c	2005-06-14 15:08:46.000000000 +0200
@@ -974,7 +974,7 @@
 	cl_signal_set_interrupt(SIGALRM, TRUE);
 	setmsrepeattimer(WAITALARM); /* Might as well be persistent ;-) */
 	while((pid=wait3(&status, waitflags, NULL)) > 0
-	||	errno == EINTR) {
+	||	(pid < 0 && errno == EINTR)) {
 		cancelmstimer();
 		if (pid > 0) {
 			++childcount;
@@ -988,6 +988,12 @@
 	if (pid < 0 && errno != ECHILD) {
 		cl_perror("%s: wait3() failed"
 		,	__FUNCTION__);
+#if defined(DEBUG)
+		if (childcount < 1 && ANYDEBUG) {
+			cl_log(LOG_DEBUG, "%s called without children to wait on"
+			,	__FUNCTION__);
+		}
+#endif
 	}
 	if (alarm_count) {
 		cl_log(LOG_ERR
